### PR TITLE
Users/hnguyen/5/basic api endpoints

### PR DESCRIPTION
**Pull request by chiashenglin0204**
Issue #5 : Finish user CRUD endpoints

### CODE REVIEW

File: server/routes/route-helpers.ts/users-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+export const createUser = async (req: any, res: any) => {
+  await client
+    .query(
+      `
+        INSERT INTO users ("userName", "firebaseAuthId", "profileImgUrl")
+        VALUES ('${req.body.userName}', '${req.body.firebaseAuthId}', '${req.body.profileImgUrl}');
+      `
+    )
+    .then(() => res.send("you have successfully created a user"))
```
Comment (hungnguyen3) [inline server/routes/route-helpers.ts/users-helpers.ts]: change this to `res.status(200).send("you have successfully created a user")`.

to have a status code for whether it is successful or not. 200 is a standard code for success.

Comment (hungnguyen3) [reply server/routes/route-helpers.ts/users-helpers.ts]: [check out this link](https://www.dotcom-monitor.com/blog/2022/03/03/the-10-most-common-http-status-codes/#:~:text=Status%20Code%20404%20%E2%80%93%20The%20most,due%20to%20improper%20URL%20redirection.)

File: server/routes/route-helpers.ts/users-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  await client
+    .query(
+      `
+			DELETE FROM users WHERE "userId" = '${req.params.userId}';
+		`
+    )
+    .then(() =>
+      res.send("you have successfully deleted a user uid: " + req.params.userId)
+    )
+    .catch((e: any) => res.json({ error: e }));
```
Comment (hungnguyen3) [inline server/routes/route-helpers.ts/users-helpers.ts]: same here, `res.status(404).json({error: e})`

read more about what 404 error is [here](https://www.dotcom-monitor.com/blog/2022/03/03/the-10-most-common-http-status-codes/#:~:text=Status%20Code%20404%20%E2%80%93%20The%20most,due%20to%20improper%20URL%20redirection.)

also other places also need status codes too. I won't put a comment in all of them.

File: server/routes/route-helpers.ts/users-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+export const deleteUserById = async (req: any, res: any) => {
+  await client
+    .query(
+      `
+			DELETE FROM users WHERE "userId" = '${req.params.userId}';
+		`
+    )
+    .then(() =>
+      res.send("you have successfully deleted a user uid: " + req.params.userId)
```
Comment (hungnguyen3) [inline server/routes/route-helpers.ts/users-helpers.ts]: status code 200

**Review by hungnguyen3 — CHANGES_REQUESTED** [review]

**Comment by chiashenglin0204** [comment]
Please review my changes for status code, I'm not sure the status code for the error with rowCount > 1 but I put 404 for now

**Comment by hungnguyen3** [comment]
> Please review my changes for status code, I'm not sure the status code for the error with rowCount > 1 but I put 404 for now

mostly looks good! However, it seems like you misunderstand the status code `400`. We can discord later and I will explain this!

**Review by hungnguyen3 — APPROVED** [review]

**Pull request merged**